### DISCUSSION TITLE
Image source rotation on ARTPEC-7 cameras

### DISCUSSION THREAD

**Discussion by jcbeaulieu** [AxisCommunications/discussions#216] [VAPIX] [answered]
Hi,

We have an issue on a AXIS P3247-LV and AXIS P1455-LE on Axis OS 10.4.0, where the image rotation setting is sometimes reset to 0 degrees.

We've been told by support that ARTPEC-7 cameras and newer no longer support video rotation at the image source,
which we assume would be `/axis-cgi/param.cgi?action=update&ImageSource.I0.Rotation=0`

But it seems that the two calls we use to set the rotation still work most of the time with these cameras.

The calls we use:

HTTP (returns 200 ok):
`/param.cgi?action=update&Image.I0.Appearance.Rotation=180`

In RTSP SETUP and PLAY
`rtsp://<IP>:554/axis-media/media.amp/stream=0?videocodec=h265&audio=0&resolution=2592x1944&camera=1&compression=30&rotation=180&...`

In the server logs, we can see that the camera logs an error on what could be a video rotation call.  
`[ ERR     ] parhand[709]: serveSetRequest: Set request with unsupported or wrong data type for root.Layout.DefaultVideoFormat.`
We then get multiple other errors immediately after:
`[ ERR     ] respawnd[25554]: execv failed!`
`[ ERR     ] video-scene-provider: Cannot grab image`
`[ ERR     ] video-scene-provider: Failed fetch scene`
`[ ERR     ] video-scene-provider: Failed running MOTE`

We currently assume video rotation is supported using the "Properties.Image.Rotation=0,90,180,270" parameter, which does return a value in our case.

Should we use a different parameter to determine if the video rotation is supported by the camera?
Are the `Image.I0.Appearance.Rotation` and RTSP `rotation=` calls still supported?
If not, should we switch to using `/axis-cgi/videooutput/setrotation.cgi?<argument>=<value>` and is this supported on very old cameras (Axis OS 5.xx)?

Thank you

**Comment by shadysp** [comment, marked as answer]
- Yes as you mentioned  use [ set rotation ](https://www.axis.com/vapix-library/subjects/t10175981/section/t10110390/display?section=t10110390-t10111694)from the [Video Output API.](https://www.axis.com/vapix-library/subjects/t10175981/section/t10110390/display)

Property

    Properties.VideoOutputControl.VideoOutputControl=yes
Property

    Properties.VideoOutputControl.OutputType=[Output type]
Firmware

    6.10 and later


-------------------------------------------


- The image rotation is hardware related.
The ImageSource parameter was introduced with ARTPEC-7 chip products to rotate the image for all image streams. As with ARTPEC-7 and later chips it is not possible to rotate  one stream only.



Property

    ImageSource.I#.SourceRotation=yes
Firmware

    8.40 and later


See VAPIX library[ Image source rotation](https://www.axis.com/vapix-library/subjects/t10175981/section/t10135583/display)

"Historically, Axis products received a native stream with a default rotation value where each video source were able to have their own user-defined rotation value. This means that if the native rotation was 0 degrees internally, the outgoing video stream could show up at either 90 or 180 degrees instead, but only as long as they had been appropriately configured. This does not hold true for newer products with auto-rotation, as they use global parameters to control the rotation of the image source. Thus, depending on what has been globally set, the video source and video stream will both have a defined rotation value, making it impossible to request a video stream with a certain rotation value when it differs from the global configuration value."

<p></p>

Parameter | Valid values | Description
-- | -- | --
ImageSource.I#.SourceRotation | yes no | Allows the client to determine whether or not a given product has image source rotation. This parameter is read-only.
Properties.Image.Rotation | [0, 90, 180, 270] | The rotation of the source.
ImageSource.I#.AutoRotationEnabled | yes no | In products with AutoRotation it is possible to activate it by setting ImageSource.I#.AutoRotationEnabled to ‘yes’. AutoRotationEnabled can be disabled by setting Rotation.
